### PR TITLE
rm -rf disco

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-frontend/issues/9379

---

This patch removes everything related to the "disco" app. There is still one or two comments mentioning "Discopane" in the code, but I am not sure what we should be doing so I filed https://github.com/mozilla/addons-frontend/issues/9805.

It's safe to remove the code because we don't build and deploy the disco app anymore. There is still a version up and running in all envs to avoid breaking old Firefox (although that wouldn't break anything really).

### CODE REVIEW

**Comment by willdurand** [comment]
The failure in the `pr` build seems normal. We removed the `flake8` job so there is nothing to execute. I am not sure why this job is still executed, though.

File: README.md [hunk 1]
```diff
@@ -404,7 +386,6 @@ curl https://addons-dev.allizom.org/__version__
 
 This project will hold distinct front-ends e.g:
```
Comment (eviljeff) [inline README.md:387]: nit: this section doesn't make sense now there is only one front-end but meh

File: bin/start-func-test-server.js [hunk 2]
```diff
@@ -1,198 +0,0 @@
-/* @flow */
```
Comment (eviljeff) [inline bin/start-func-test-server.js:1]: this file was only for disco?  (couldn't see anything particularly disco related in it, but I don't know what it any of it does 😄 )

File: package.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
   },
   "scripts": {
     "build": "npm run clean && better-npm-run build",
-    "build-all": "npm run clean && NODE_APP_INSTANCE=amo better-npm-run build && NODE_APP_INSTANCE=disco better-npm-run build",
+    "build-all": "npm run clean && NODE_APP_INSTANCE=amo better-npm-run build",
```
Comment (eviljeff) [inline package.json:12]: is it still necessary to do `NODE_APP_INSTANCE=amo` when there's only one app? (if not this is the same as `build` now)

File: package.json [hunk 4]
```diff
@@ -39,7 +37,7 @@
     "storybook-build": "better-npm-run storybook-build",
     "storybook-deploy": "storybook-to-ghpages --script storybook-build",
     "storybook-smoke-test": "better-npm-run storybook-smoke-test",
-    "test-ci": "bin/config-check.js && NODE_APP_INSTANCE=amo npm run build-locales && NODE_APP_INSTANCE=disco npm run build-locales && better-npm-run test-ci && codecov",
+    "test-ci": "bin/config-check.js && NODE_APP_INSTANCE=amo npm run build-locales && better-npm-run test-ci && codecov",
```
Comment (eviljeff) [inline package.json:40]: similar question about the need for `NODE_APP_INSTANCE`

File: src/core/client/base.js [hunk 5]
```diff
@@ -81,9 +81,6 @@ export default async function createClient(
       // When `forceRefresh` is `true`, client side navigation is disabled and
       // all links will trigger a full page reload. This is what we want when
       // the server has loaded an anoynmous page.
-      //
-      // Note: we check the presence of the `site` state because the `disco`
-      // app uses the same code but does not use this state.
       forceRefresh:
         initialState && initialState.site
           ? initialState.site.loadedPageIsAnonymous
```
Comment (eviljeff) [inline src/core/client/base.js:86]: seems like we won't need the check on `.site` then?

**Review by eviljeff — APPROVED** [review]
🔥

File: bin/start-func-test-server.js (hunk 2 above)
Comment (willdurand) [reply bin/start-func-test-server.js:1]: It was only used for the ui tests on travis-ci, which covered the discopane only

File: package.json (hunk 3 above)
Comment (willdurand) [reply package.json:12]: it's still being used by `node-config` and other parts of our app.. I'll file an issue to try to get rid of it.

Comment (willdurand) [reply package.json:12]: https://github.com/mozilla/addons-frontend/issues/9807

Comment (eviljeff) [reply package.json:12]: `build` works without it being defined?

File: src/core/client/base.js (hunk 5 above)
Comment (willdurand) [reply src/core/client/base.js:86]: I added a note to: https://github.com/mozilla/addons-frontend/issues/9805

File: package.json (hunk 3 above)
Comment (willdurand) [reply package.json:12]: no, the readme says to invoke `build` with the env var

**Pull request merged**